spec/models/mhv_account_spec.rb

```diff
@@ -281,43 +281,20 @@
             .and not_trigger_statsd_increment('mhv.account.existed')
           expect(subject.persisted?).to be_truthy
           expect(subject.account_state).to eq('upgraded')
-          expect(subject.registered_at).to be_nil
           expect(subject.upgraded_at).to be_a(Time)
           expect(subject.eligible?).to be_truthy
           expect(subject.terms_and_conditions_accepted?).to be_truthy
         end
       end
     end
 
-    context 'existing account that has already been upgraded' do
```

Comment:
This scenario shouldn't have been possible anyway, even before my changes. If an account is in the `upgraded` state, `may_upgrade` should return `false` for the conditional block where `mhv_ac_client.post_upgrade` occurs, so it shouldn't be possible to reach the `rescue` block.

Reply:
The problem is that the account could be "upgraded" / "premium" and we don't know that it is or not until we make the call to upgrade. This test was there to satisfy that particular scenario.

Reply:
So should I re-introduce this case?

Reply:
I think we should include this case, yes. Perhaps update the description to reflect what I just described if its not clear.

Reply:
Don't we know whether it's upgraded based on `previously_upgraded?` (or `upgraded_at`)?

Reply:
Wouldn't we know what it is based on `upgraded_at` (and thus `previously_upgraded`)?